docs/quickstarts/baremetal.md

```diff
@@ -0,0 +1,283 @@
+Flatcar Container Linux utilizing PXE.
+
+By the end of this guide, you'll have a working Kubernetes cluster with 1 controller node and 2
+worker nodes.
+
+## Requirements
+
+* Basic understanding of Kubernetes concepts.
+* Terraform v0.12.x, [terraform-provider-matchbox](https://github.com/poseidon/terraform-provider-matchbox)
+and [terraform-provider-ct](https://github.com/poseidon/terraform-provider-ct) installed locally.
+* Machines with 2GB RAM, 30GB disk, PXE-enabled NIC, IPMI.
+* PXE-enabled [network boot](https://coreos.com/matchbox/docs/latest/network-setup.html) environment.
+* Matchbox v0.6+ deployment with API enabled.
+* Matchbox credentials `client.crt`, `client.key`, `ca.crt`.
+* An SSH key pair for management access.
+* `kubectl` installed locally to access the Kubernetes cluster.
+
+Note that the machines should only be powered on after starting the installation, see below.
+
+## Steps
+
+### Step 1: Install lokoctl
+
+Lokoctl is a command-line interface for Lokomotive.
+
+To install `lokoctl`, follow the instructions in the [lokoctl installation](../installer/lokoctl.md)
+guide.
+
+### Step 2: Set up a working directory
+
+
+It's better to start fresh in a new working directory, as the state of the cluster is stored in this
+directory.
+
+This also makes the cleanup task easier.
+
+
+```console
+mkdir -p lokomotive-infra/mybaremetalcluster
+cd lokomotive-infra/mybaremetalcluster
+```
+
+### Step 3: Machines, DNS and Matchbox set up
+
+#### Machines
+
+Mac addresses collected from each machine.
+
+For machines with multiple PXE-enabled NICs, pick one of the MAC addresses. MAC addresses will be
+used to match machines to profiles during network boot.
+
+Example:
+
+```console
+52:54:00:a1:9c:ae (node1)
+52:54:00:b2:2f:86 (node2)
+52:54:00:c3:61:77 (node3)
+```
+
+#### DNS
+
+Create DNS A (or AAAA) record for each node's default interface.
+
+Cluster nodes will be configured to refer to the control plane and themselves by these fully
+qualified names and they will be used in generated TLS certificates.
+
+Example:
+
+```console
+node1.example.com (node1)
+node2.example.com (node2)
+node3.example.com (node3)
+```
+
+#### Matchbox
+
+One of the requirements is to have Matchbox with TLS enabled deployed.
+
+Verify the matchbox read-only HTTP endpoints are accessible.
+
+```console
+curl http://matchbox.example.com:8080
+matchbox
+```
+
+Verify your TLS client certificate and key can be used to access the Matchbox API.
+
+```console
+openssl s_client -connect matchbox.example.com:8081 \
+  -CAfile /path/to/matchbox/ca.crt \
+  -cert /path/to/matchbox/client.crt \
+  -key /path/to/matchbox/client.key
+```
+
+
+### Step 4: Define Cluster Configuration
+
+To create a Lokomotive cluster, we need to define a configuration.
+
+Create a file with the extension `.lokocfg` with the below contents:
+
+```tf
+# baremetalcluster.lokocfg
+cluster "bare-metal" {
```

Comment:
We could convert that to `examples` configuration too.

Reply:
I think or baremetal it wouldn't make sense as there will be lot of variations for confgurations.